pkg/chain/upkeep_observation.go

```diff
@@ -0,0 +1,10 @@
+package chain
+
+import (
+	"github.com/smartcontractkit/ocr2keepers/pkg/types"
+)
+
+type UpkeepObservation struct {
```

Comment:
I had to add this in chain as it needed a concrete type for BlockKey. Followed the same pattern as BlockKey and UpkeepKey.

I found this very confusing as to why we needed different interface and concrete type and the need to convert between each other in code. Also these types are in chain package which is confusing by itself, since some files here like evm registry is not actually used, but the concrete types are actually used in prod code. Would like to discuss how we can structure this better

Reply:
This is part of the design that needs to be revisited. I already have a working theory on how to do this so we can have that discussion whenever.

Reply:
also wasn't able to find an easy way other than copy the type in CL repo https://github.com/smartcontractkit/chainlink/pull/8457 . Definitely need to fix this before merging

Reply:
created a todo for now, don't want to block this PR since this pattern is already present in codebase